launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/LoginServlet.java

```diff
@@ -37,16 +51,43 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     if (userService.isUserLoggedIn()) {
       String userEmail = userService.getCurrentUser().getEmail();
       String logoutUrl = userService.createLogoutURL(urlToRedirectTo);
-
       String loginMessage = "<p>Logged in as " + userEmail + ". <a href=\"" + logoutUrl + "\">Logout</a>.</p>";
-      LoginStatus loginStatus = new LoginStatus(true, loginMessage);
+
+      Query query =
+        new Query(LoginStatus.USER_FEED_KEY).addSort(LoginStatus.TIMESTAMP_KEY, SortDirection.DESCENDING);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+
+      ArrayList<UserFeed> userFeeds = new ArrayList<UserFeed>();
+      for (Entity entity : results.asIterable()) {
+        String userFeedEmail = String.valueOf(entity.getProperty(LoginStatus.EMAIL_KEY));
+        if (userEmail.equals(userFeedEmail)) {
+          String title = (String) entity.getProperty(LoginStatus.TITLE_KEY);
+          String name = (String) entity.getProperty(LoginStatus.NAME_KEY);
+          String description = (String) entity.getProperty(LoginStatus.DESCRIPTION_KEY);
+          String email = (String) entity.getProperty(LoginStatus.EMAIL_KEY);
+          long timestamp = (long) entity.getProperty(LoginStatus.TIMESTAMP_KEY);
+          Date date = new Date(timestamp);
+          SimpleDateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy  HH:mm:ss Z", Locale.getDefault());
+          String postTime = dateFormat.format(date);
+          Key key = entity.getKey();
+          
+          String urlID = KeyFactory.keyToString(entity.getKey()); // the key string associated with the entity, not the numeric ID.
+          String rssLink = BASE_URL + urlID;
+
+          userFeeds.add(new UserFeed(title, name, rssLink, description, email, postTime, key));
+        }
+      }
+      
+      LoginStatus loginStatus = new LoginStatus(true, loginMessage, userFeeds);
 
       response.getWriter().println(GSON.toJson(loginStatus));
     } else {
       String loginUrl = userService.createLoginURL(urlToRedirectTo);
-
       String loginMessage = loginUrl;
-      LoginStatus loginStatus = new LoginStatus(false, loginMessage);
+
+      LoginStatus loginStatus = new LoginStatus(false, loginMessage, new ArrayList<UserFeed>());
```

Comment:
nit: it would be better to have static construction methods in LoginStatus (e.g. forFailedLogin(loginMessage) and forSuccess(loginMessage, userFeeds)). that way you don't have to pass in true/false and an empty list to the constructor - those static constructions methods do that within their method body

Reply:
done.